presentation/src/main/java/com/example/user/test/presentation/base/BaseActivity.kt

```diff
@@ -0,0 +1,43 @@
+package com.example.user.test.presentation.base
+
+import android.os.Bundle
+import android.os.PersistableBundle
+import android.support.v7.app.AppCompatActivity
+import android.util.Log
+import com.arellomobile.mvp.MvpAppCompatActivity
+import com.arellomobile.mvp.MvpPresenter
+import com.arellomobile.mvp.presenter.InjectPresenter
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.disposables.Disposable
+
+abstract class BaseActivity<P: MvpPresenter<*>, R: BaseRouter<*>>: MvpAppCompatActivity() {
+
+    protected val router: R
+
+    init {
+        router = provideRouter()
+    }
+
+    private val compositeDisposable: CompositeDisposable by lazy {
+        CompositeDisposable()
+    }
+
+    abstract fun provideLayoutId(): Int
+
+    abstract fun provideRouter(): R
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        Log.d("myLog", "CharactersActivity.onCreate")
```

Comment:
If you pass logs into a repository, it must log only for `DEBUG` version, else delete it.